src/main/scala/Exercise8_1.scala

```diff
@@ -0,0 +1,28 @@
+object Exercise8_1 extends App {
+  class BankAccount(initialBalance: Double) { // provided in the exercise 8.1
+    private var balance = initialBalance // I cannot access or override it in subclasses!
+    def currentBalance = balance
+    def deposit(amount: Double) = {
+      balance += amount;
+      balance
+    }
+    def withdraw(amount: Double) = {
+      balance -= amount;
+      balance
+    }
+  }
+  class CheckingAccount(initialBalance: Double) extends BankAccount(initialBalance) { // this is the exercise
+    val bankCommission = 1.0
```

Comment:
you should make this `private` since this class isn't `final` and anyone is able to extend it and override it. (Unless this is the purpose of the exercise, check it out)

As I said, I don't know the exercise (a good practice would be a few lines when you open a pull request, explaining the purpose of it) but you might also pass `bankCommission` as a class parameter, in order to make different commissions when you instance it.

Everything is technically ok, it's just that it depends on the purpose of the exercise. I've only mentioned this to you to know it.

Reply:
When I created this class, I wanted bankCommission to be accessible with a getter, but not to be changeable, because the exercise provided me a fixed value. In that case, anyone could extend this class and override it so as to make a different account with a different commission.

Reply:
When I created this class, I wanted bankCommission to be accessible with a getter, but not to be changeable, because the exercise provided me a fixed value. In that case, anyone could extend this class and override it so as to make a different account with a different commission.

Reply:
From now on, I will include the exercise heading at the beginning of each file as a comment.